ocaml/xapi/message_forwarding.ml

```diff
@@ -161,25 +161,29 @@ let log_exn_ignore ?(doc = "performing unknown operation") f x =
 
 
 (* Given an SR, return a PBD to use for some storage operation. *)
-(* In the case of SR.destroy we need to be able to forward the SR operation when all
-   PBDs are unplugged - this is the reason for the consider_unplugged_pbds optional
-   argument below. All other SR ops only consider plugged PBDs... *)
-let choose_pbd_for_sr ?(consider_unplugged_pbds=false) ~__context ~self () =
+(* In the case of SR.destroy and forget we need to be able to forward the SR
+   operation when all PBDs are unplugged. This is the reason for the
+   consider_unplugged_pbds optional argument below. All other SR ops only
+   consider plugged PBDs. *)
+let choose_pbds_for_sr ?(consider_unplugged_pbds=false) ~__context ~self () =
   let all_pbds = Db.SR.get_PBDs ~__context ~self in
-  let plugged_pbds = List.filter (fun pbd->Db.PBD.get_currently_attached ~__context ~self:pbd) all_pbds in
+  let plugged_pbds = List.filter (fun self -> Db.PBD.get_currently_attached ~__context ~self) all_pbds in
```

Comment:
perhaps write a helper for this? looks like it is used in multiple places, e.g. 

https://github.com/xapi-project/xen-api/blob/74412d584c9e49669256c3243ccdbdb1a84eff0c/ocaml/xapi/agility.ml#L28

Reply:
Not sure it's worth doing it in this PR, it's a pattern that's repeated a lot in xapi. Could be tackled all in a another PR where it can be treated more systematically, it might need more considerations.